modules/internal/ArrayViewRankChange.chpl

```diff
@@ -470,6 +762,75 @@ module ArrayViewRankChange {
     proc _getRCREView() {
       return this;
     }
+
+    proc dsiDestroyArr(isalias:bool) {
+      if ownsArrInstance {
+        _delete_arr(_ArrInstance, _isPrivatized(_ArrInstance));
+      }
+    }
+  }  // end of class ArrayViewRankChangeArr
+
+
+  // helper routines for converting things from higher- to
+  // lower-dimensions and back again
+
+  inline proc chpl_rankChangeConvertIdx(i: integral, collapsedDim, idx) {
+    param downrank = collapsedDim.size;
+    var ind = idx;
+    var j = 1;
+    for param d in 1..downrank {
+      if !collapsedDim(d) {
+        ind(d) = i;
+        j += 1;
+      }
+    }
+    assert (j == 2);
```

Comment:
Yes, this is the i-isn't a tuple case and I believe it's called for 1D cases in which idxType is a scalar rather than a 1-tuple.

I can look into refactoring the conversion routines to make better use of each other.